packages/lib-classifier/src/plugins/tasks/Registry.js

```diff
@@ -0,0 +1,29 @@
+class Registry {
+  constructor () {
+    this.register = {}
+  }
+
+  get (key) {
+    return this.register[key] || {}
+  }
+
+  add (key, value) {
+    if (this.register[key]) {
+      throw new Error(`Registry error: key ${key} is already registered.`)
```

Comment:
This is a bit of a wild idea since I've never used this myself in production code, but how about using a set which enforces uniqueness? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set

Support across browsers looks pretty good.